src/items/use-declarations.md

```diff
@@ -201,8 +315,88 @@ m!(use std as _;);
 // use std as _;
 ```
 
-[IDENTIFIER]: ../identifiers.md
+## Restrictions
+
+The following are restrictions for valid `use` declarations.
+
+* `use crate;` must use `as` to define the name to bind the crate root to.
+* `use {self};` is an error, there must be a leading segment when using `self`.
+* As with any item definition, `use` imports cannot create duplicate bindings of the same name in the same namespace in a module or block.
+* `use` paths with `$crate` are not allowed in a [`macro_rules`] expansion.
+* `use` paths cannot refer to enum variants through a [type alias]. Example:
+  ```rust,compile_fail
+  enum MyEnum {
+      MyVariant
+  }
+  type TypeAlias = MyEnum;
+
+  use MyEnum::MyVariant; // OK
+  use TypeAlias::MyVariant; // ERROR
+  ```
+
+## Ambiguities
+
+> **Note**: This section is incomplete.
+
+Some situations are an error when there is an ambiguity as to which name a `use` declaration refers to, when there are two name candidates that do not resolve to the same entity.
+
+Glob imports are allowed to import conflicting names in the same namespaces as long as the name is not used or shadowed.
```

Comment:
What's meant by *or shadowed* here?  E.g., this works:

```rust
mod foo {
    pub struct Qux;
}

mod bar {
    pub struct Qux;
}

use bar::*;
use foo::*; // Ok, no name conflict.

struct Qux; // Shadows imports.

fn main() {}
```

Reply:
Oof, I don't remember. It's possible I meant the case exactly what you specify is OK. To reword it as "...as long as the name is shadowed or not used"? However, that doesn't really make much sense (it isn't "used" when it is shadowed, either).

Another possibility is that I got `AmbiguityKind::GlobVsOuter` confused here. But I believe that restriction is a bit more complicated. See these examples:

https://github.com/rust-lang/rust/blob/HEAD/tests/ui/imports/glob-shadowing.rs
https://github.com/rust-lang/rust/blob/HEAD/tests/ui/imports/issue-57539.rs
https://github.com/rust-lang/rust/blob/HEAD/tests/ui/imports/shadow_builtin_macros.rs
https://github.com/rust-lang/rust/blob/HEAD/tests/ui/macros/macro-path-prelude-shadowing.rs

Perhaps it is best to just delete "or shadowed" for now, and follow up on completing this section later?

Reply:
Done.